compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -2394,7 +2400,13 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 ty::Const::new_unevaluated(tcx, ty::UnevaluatedConst::new(did, args))
             }
             Res::Def(DefKind::AssocConst, did) => {
-                debug_assert!(path.segments.len() >= 2);
+                if path.segments.len() < 2 {
+                    let guar = self
+                        .dcx()
+                        .struct_span_err(span, "cannot infer type, type annotations needed")
```

Comment:
```suggestion
                        .struct_span_err(span, "cannot infer value")
```
from what I can tell of [`tests/ui/use/import_trait_associated_item_bad.rs`](https://github.com/rust-lang/rust/pull/138712/files#diff-0e2aa8759cbed2eb814cd98817fe4add695262c07791bd175cea0e1d3bfd5bcdR14) the  error isn't really type related?

https://github.com/rust-lang/rust/pull/138712/files#diff-0e2aa8759cbed2eb814cd98817fe4add695262c07791bd175cea0e1d3bfd5bcdR14

Reply:
The error is copypasted from the non-`min_generic_const_args` case, see #138711.
(It's the `Self` type that cannot be inferred here.)